apps/antalmanac/src/stores/HoveredStore.ts

```diff
@@ -1,33 +1,64 @@
 import { create } from 'zustand';
-import { AASection } from '@packages/antalmanac-types';
-import { calendarizeCourseEvents } from './calendarizeHelpers';
+import { AASection, ScheduleCourse } from '@packages/antalmanac-types';
+import { calendarizeCourseEvents, calendarizeFinals } from './calendarizeHelpers';
 import { CourseEvent } from '$components/Calendar/CourseCalendarEvent';
 import { CourseDetails } from '$lib/course_data.types';
 
+const HOVERED_SECTION_COLOR = '#80808080';
 export interface HoveredStore {
-    hoveredCourseEvents: CourseEvent[] | undefined;
-    setHoveredCourseEvents: (section?: AASection, courseDetails?: CourseDetails, term?: string) => void;
+    hoveredEvents: ScheduleCourse[] | undefined;
+    setHoveredEvents: (section?: AASection, courseDetails?: CourseDetails, term?: string) => void;
+    hoveredCalendarizedCourses: CourseEvent[] | undefined;
+    hoveredCalendarizedFinal: CourseEvent | undefined;
 }
 
+const DEFAULT_HOVERED_STORE = {
+    hoveredEvents: undefined,
+    hoveredCalendarizedCourses: undefined,
+    hoveredCalendarizedFinal: undefined,
+};
+
 export const useHoveredStore = create<HoveredStore>((set) => {
     return {
-        hoveredCourseEvents: undefined,
-        setHoveredCourseEvents: (section, courseDetails, term) => {
-            set({
-                hoveredCourseEvents:
-                    section && courseDetails && term
-                        ? calendarizeCourseEvents([
-                              {
-                                  ...courseDetails,
-                                  section: {
-                                      ...section,
-                                      color: '#80808080',
-                                  },
-                                  term,
-                              },
-                          ])
-                        : undefined,
-            });
+        ...DEFAULT_HOVERED_STORE,
+        setHoveredEvents: (section, courseDetails, term) => {
+            if (section == null || courseDetails == null || term == null) {
+                set({ ...DEFAULT_HOVERED_STORE });
+                return;
+            } else {
```

Comment:
```suggestion
            }
```
If you have a guard clause, you can remove the explicit `else`, since there's no way for the code to continue after reaching a return statement.